### PR TITLE
Synthetic PoRep

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

**Comment by jennijuju** [comment]
should we consider developing in a feature branch?

**Comment by snadrus** [comment]
@jennijuju This adds functions that are currently uncalled.

**Comment by cryptonemo** [comment]
> should we consider developing in a feature branch?

Do you mean on the lotus side?  If so, I would think yes :+1: 

Or do you mean renaming the existing branch in this repo?

**Comment by cryptonemo** [comment]
> @jennijuju This adds functions that are currently uncalled.

Which ones?  It shouldn't :thinking:

**Comment by snadrus** [comment]
No Lotus code calls the Synthetic PoREP functions yet.
The new functions are called by tests.

**Comment by cryptonemo** [comment]
> No Lotus code calls the Synthetic PoREP functions yet. The new functions are called by tests.

Correct, sorry I misunderstood the question :sweat_smile:

File: rust/src/fvm/machine.rs [hunk 1]
```diff
@@ -378,6 +382,8 @@ pub struct TraceMessage {
     #[serde(with = "strict_bytes")]
     pub params: Vec<u8>,
     pub codec: u64,
+    pub gas_limit: u64,
+    pub read_only: bool,
```
Comment (Stebalien) [inline rust/src/fvm/machine.rs:386]: Looks like this needs a rebase?

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
This should be good to go. The next step is to:

1. Merge it.
2. Forward port to master.

**Pull request merged**